addons/mail/models/mail_channel.py

```diff
@@ -489,6 +489,19 @@ def _message_post_after_hook(self, message, msg_vals):
         self._set_last_seen_message(message)
         return super()._message_post_after_hook(message=message, msg_vals=msg_vals)
 
+    @api.model
+    def _message_edit_content(self, message, body):
+        super()._message_edit_content(message, body)
+        payload = {
+            'id': message.id,
+            'body': body,
+        }
+        self.env['bus.bus'].sendone((self._cr.dbname, 'mail.channel', message.res_id), {
+            'type': 'mail.message_update',
+            'payload': payload,
+        })
+        return payload
```

Comment:
It's strange to have an override method change the API. and return something that the parent didn't. I would make the original method return it instead.

Reply:
I think this could be moved to the controller itself. Method does business update / send notifications, controller builds the payload. That way no strange value returned on model, easier override, and controller adds its own data manipulation.

Reply:
If I build the payload in the controller I need to pass it into the method anyway because the notification needs it, but the mail.thread method doesn't care about the payload so it'll be forced to take it as an argument and ignore it anyway, just so that its override can use it (unless I adapt the method call depending on the model but at that point we're back to square one...)

I guess I can just build the "payload" twice, once in the controller and once in the mail.channel overload so that I can send it as notification.

Reply:
I think we are over complicating it, just return it in the main method.
It's especially necessary because the chatter will not receive the bus notification (as not a channel) but it needs to receive the payload from the RPC.

Reply:
As a reminder the route is just there so we can grant access to guests, everything except that should be done in models.

Reply:
Both are fine to me. Main point according to me is to have true multi methods with clean overrides (and avoid if model == "channel"). Remaining is more a popotte.